src/lightning/data/streaming/dataset.py

```diff
@@ -257,73 +289,34 @@ def __next__(self) -> Any:
         self.global_index += 1
         self.index += 1
 
-        # Checkpoint based on time
-        if self.checkpoint_interval and (time() - self.last_time) > self.checkpoint_interval:
-            self._checkpoint(self.chunk_index - 1)
-
         return data
 
-    def _checkpoint(self, chunk_index: int) -> None:
-        if self.checkpoint_interval is None:
-            return
-
-        if not _is_in_dataloader_worker():
-            return
-
-        assert self.cache
-        assert self.worker_env
-
-        with tempfile.TemporaryDirectory() as tmpdir:
-            tmp_checkpoint_path = os.path.join(tmpdir, "checkpoint.json")
-            with open(tmp_checkpoint_path, "w") as f:
-                # 1. Write the state to a tempfile
-                json.dump(
-                    {
-                        "rank": self.cache._reader.rank,
-                        "current_epoch": self.current_epoch,
-                        "input_dir_path": self.input_dir.path,
-                        "input_dir_url": self.input_dir.url,
-                        "item_loader": self.item_loader.state_dict() if self.item_loader else None,
-                        "drop_last": self.drop_last,
-                        "seed": self.seed,
-                        "checkpoint_interval": self.checkpoint_interval,
-                        "chunk_index": chunk_index,
-                        "global_index": self.global_index,
-                        "index": self.index,
-                        "world_size": self.distributed_env.world_size,
-                        "num_workers": self.worker_env.world_size,
-                        "shuffle": self.shuffle,
-                    },
-                    f,
-                )
-
-            # 4. Move the file to its target position
-            shutil.move(tmp_checkpoint_path, os.path.join(self.cache.checkpoint_dir, "checkpoint.json"))
-
-        self.last_time = time()
-
-    def state_dict(self, num_samples_yielded: int = 0, num_workers: int = 0, batch_size: int = 1) -> Dict[str, Any]:
+    def state_dict(self, num_samples_yielded: int, num_workers: int, batch_size: int) -> List[Dict[str, Any]]:
```

Comment:
Same here, a dict would be desired here. If you need to store things in a list, we could do:

```py
return {"the_name_of_the_list_thing": [...]} 
```